tests/functional/test_supported_protocols.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import pytest
+
+from botocore.args import PRIORITY_ORDERED_SUPPORTED_PROTOCOLS
+from botocore.session import get_session
+
+
+def _all_test_cases():
+    session = get_session()
+    loader = session.get_component('data_loader')
+
+    services = loader.list_available_services('service-2')
+    services_models_with_protocols = []
+    services_models_without_protocols = []
+
+    for service in services:
+        service_model = session.get_service_model(service)
+        if 'protocols' in service_model.metadata:
+            services_models_with_protocols.append(service_model)
+        else:
```

Comment:
Do we have models that don't have `protocols` entries currently? If not, should this just be an error case?

Reply:
It would break custom models, no?

Reply:
I don't know if we're intending to test custom models? Since this isn't in the runtime path, if it does break, I think the fix at that point is simple and ensures the quality of the model improves to match what we expect in production.

Reply:
I think this also means we never run the second test?

Reply:
My bad, I misread which line the comment was referring to from GH mobile.  

Yes, we have models that don't have `protocols`.  148 at the time of writing this comment.